### PR TITLE
Add interactive UI documentation

### PR DESCRIPTION
**Pull request by ziad-saab**

### CODE REVIEW

**Comment by Montoya** [comment]
How do we get another preview published?

**Review by Montoya — APPROVED** [review]

**Comment by Montoya** [comment]
This is ready for review now

**Review by alexandratran — CHANGES_REQUESTED** [review]

File: snaps/features/custom-ui.md [hunk 1]
```diff
@@ -85,6 +116,37 @@ const content = panel([
...
+const form = form({
+  name: 'form-to-fill',
+  children: [
+    input({
+      name: 'user-name',
+      placeholder: 'Your name',
+    }),
+    button({
+      value: 'Submit!',
+      type: 'submit'
```
Comment (alexandratran) [inline snaps/features/custom-ui.md:144]: Is the parameter `type` or `buttonType`?

File: snaps/features/custom-ui.md [hunk 2]
```diff
@@ -117,6 +179,32 @@ import { image } from '@metamask/snaps-sdk';
...
+- `label`: Optional, a string that is displayed alongside the input to label it.
+- `value`: Optional, the default value of the input.
+
+#### Example
+
+```js
+import { input } from '@metamask/snaps-sdk';
+
+const amountInput = input({
+  type: 'number',
```
Comment (alexandratran) [inline snaps/features/custom-ui.md:202]: Is the parameter `type` or `inputType`?

File: snaps/reference/snaps-api.md [hunk 3]
```diff
@@ -10,9 +10,52 @@ import TabItem from '@theme/TabItem';
 # Snaps API
 
 Snaps can communicate with and modify the functionality of MetaMask using the [Snaps API](../learn/about-snaps/apis.md#snaps-api).
-To call each method, you must first [request permission](../how-to/request-permissions.md) in the Snap
+To call some of these methods, you must first [request permission](../how-to/request-permissions.md) in the Snap
```
Comment (alexandratran) [inline snaps/reference/snaps-api.md:13]: To clarify, the three new `Interface` methods do NOT require requesting permission in `initialPermissions`, but the rest of the methods do?

**Comment by alexandratran** [comment]
Made some edits for style and clarity. See my previous comments for outstanding questions.

File: snaps/features/custom-ui/index.md [hunk 4]
```diff
@@ -176,7 +280,64 @@ module.exports.onHomePage = async () => {
...
+
+An object containing:
+
+- `name`: `string` - The name that will be used as a key to the event sent to
+  [`onUserInput`](../../reference/entry-points.md#onuserinput) when the containing form is submitted.
+- `inputType`: `string` - (Optional) Type of input.
+  Possible values are `text`, `number`, or `password`.
+  The default is `text`.
+- `placeholder`: `string` - (Optional) The text displayed when the input is empty.
+- `label`: `string` (Optional) The text displayed alongside the input to label it.
```
Comment (alexandratran) [inline snaps/features/custom-ui/index.md:303]: I tried testing the `label` parameter but a label did not show up for me. The [example package](https://github.com/MetaMask/snaps/blob/main/packages/examples/packages/interactive-ui/src/index.ts) also doesn't seem to have a `label` example.

File: snaps/features/custom-ui.md (hunk 1 above)
Comment (GuillaumeRx) [reply snaps/features/custom-ui.md]: Should be `buttonType`

File: snaps/features/custom-ui.md (hunk 2 above)
Comment (GuillaumeRx) [reply snaps/features/custom-ui.md]: Should be `inputType`

File: snaps/reference/snaps-api.md (hunk 3 above)
Comment (GuillaumeRx) [reply snaps/reference/snaps-api.md:13]: `snap_createInterface`, `snap_updateInterface`, `snap_getInterfaceState` and the `onUserInput` handler are all permitted meaning there's no permission needed to use those.

File: snaps/features/custom-ui/index.md (hunk 4 above)
Comment (ziad-saab) [reply snaps/features/custom-ui/index.md:303]: cc. @GuillaumeRx

File: snaps/features/interactive-ui.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+The first step to use interactive UI is to create an interface. This is accomplished with the [`snap_createInterface`](../reference/snaps-api.md#snap_createinterface) method. This method will return the ID of the created interface. This ID can be passed to [`snap_dialog`](../reference/snaps-api.md#snap_dialog), returned from [`onTransaction`](../reference/entry-points.md#ontransaction), or from [`onHomePage`](../reference/entry-points.md#onhomepage).
+
+If you need to [update the interface](#update-an-interactive-interface) or [get its state](#get-an-interactive-interfaces-state) at a future time, you should store its ID in the Snap's storage.
+
+## Update an interactive interface
+
+To update an interactive interface that is still active, use the [`snap_updateInterface`](../reference/snaps-api.md#snap_updateinterface) method. Pass it the ID of the interface to be updated, and the new UI.
+
+Updating an interface can be done either as part of [`onUserInput`](../reference/entry-points.md#onuserinput), or as part of an asynchronous process.
```
[the remark this answers is not in the document]
Comment (ziad-saab) [reply snaps/features/interactive-ui.md]: thanks i'll add it back in

File: snaps/features/custom-ui/index.md (hunk 4 above)
Comment (ziad-saab) [reply snaps/features/custom-ui/index.md:303]: The `label` property seems to still be in the code, I wonder if it's being used or not

https://github.com/MetaMask/snaps/blob/main/packages/snaps-sdk/src/ui/components/input.ts#L47

**Review by alexandratran — APPROVED** [review]
Additional minor edits and rearranged interactive UI methods in the Snaps API reference to clarify that those methods don't require requesting permission in the manifest file. LGTM

**Comment by ziad-saab** [comment]
Still waiting on an answer about `label` before merging

File: snaps/features/custom-ui/interactive-ui.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+# Interactive UI
+
+:::flaskOnly
+:::
+
+You can display interactive user interface (UI) components.
+Interactive UI is an extension of [custom UI](index.md).
+It allows interfaces returned from [`snap_dialog`](../../reference/snaps-api.md#snap_dialog),
+[`onTransaction`](../../reference/entry-points.md#ontransaction), and
+[`onHomePage`](../../reference/entry-points.md#onhomepage) to respond to user input.]
```
Comment (Montoya) [inline snaps/features/custom-ui/interactive-ui.md:17]: Typo here, trailing ]

**Review by Montoya — DISMISSED** [review]

File: snaps/features/custom-ui/interactive-ui.md (hunk 6 above)
Comment (ziad-saab) [reply snaps/features/custom-ui/interactive-ui.md]: fixed

**Review by GuillaumeRx — APPROVED** [review]

File: snaps/features/custom-ui/index.md (hunk 4 above)
Comment (Montoya) [reply snaps/features/custom-ui/index.md:303]: This will be supported in 11.13, the docs are correct.

**Review by Montoya — APPROVED** [review]

**Pull request merged**